src/main/java/com/duke/task/TaskList.java

```diff
@@ -0,0 +1,80 @@
+package com.duke.task;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a handler that stores manages list that
+ * contains all the task information.
+ */
+public class TaskList {
+    /**
+     * A list that stores all the task information.
+     */
+    public ArrayList<Task> tasks;
+
+    /**
+     * Instantiates a <code>TaskList</code> object with empty task list.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Instantiates a <code>TaskList</code> object with task list represnted by
+     * the <code>ArrayList</code> object.
+     *
+     * @param t the list of tasks.
+     */
+    public TaskList(ArrayList<Task> t) {
+        tasks = t;
+    }
+
+    /**
+     * Marks a Task with the given index as done.
+     *
+     * @param tindex The index of the Task in the list to be marked as done.
+     * @return The Task to be marked as done.
+     */
+    public Task markTask(int tindex) {
```

Comment:
mark task is not the most intuitive name for a method as well. could change it to markTaskAsDone or markDone